frontend/src/metabase/dashboard/containers/CopyDashboardForm.tsx

```diff
@@ -82,6 +94,16 @@ function CopyDashboardForm({
     [onValuesChange],
   );
 
+  const hasDashboardQuestions = useMemo(() => {
+    return !!originalDashboard?.dashcards.some(
+      dc => dc.card.dashboard_id !== null,
+    );
+  }, [originalDashboard]);
```

Comment:
This check disables shallow copies for any dashboard with virtual cards (text, heading, link, iframe)

Reply:
Great catch. Fixed and adjusted an e2e test to add a virtual card to ensure this works correctly.